CMakeLists.txt

```diff
@@ -91,6 +94,35 @@ if(USE_CUDA)
   add_compile_definitions(__CUDA)
 endif()
 
+if(USE_ROCM)
+  set(CMAKE_CXX_STANDARD 11) 
+  SET(CMAKE_HIP_STANDARD 11)
```

Comment:
Please note that CMake add support to `HIP` in version `3.21`. This is rather a new version.